src/neo/SmartContract/Native/Tokens/Nep17Token.cs

```diff
@@ -22,12 +22,14 @@ public abstract class Nep5Token<TState> : NativeContract
         protected const byte Prefix_TotalSupply = 11;
         protected const byte Prefix_Account = 20;
 
-        protected Nep5Token()
+        protected Nep17Token()
         {
             this.Factor = BigInteger.Pow(10, Decimals);
 
             Manifest.Features = ContractFeatures.HasStorage;
-            Manifest.SupportedStandards = new[] { "NEP-5" };
+            Manifest.SupportedStandards = new[] { "NEP-17" };
+            Manifest.Extra = new IO.Json.JObject();
+            Manifest.Extra["name"] = Name;
```

Comment:
Maybe we should add a new field in manifest. `Extra` is optional.

Reply:
done